Note this patch leaks the default GIRepository and would not be suitable for
upstream. But it's a one-time leak of global data, so who cares.

Totem upstream uses libpeas2 instead.

Patch by: Leigh Scott
This note by: Michael Catanzaro

diff -uNrp a/src/plugins/totem-plugins-engine.c b/src/plugins/totem-plugins-engine.c
--- a/src/plugins/totem-plugins-engine.c	2025-05-21 14:08:29.000000000 +0100
+++ b/src/plugins/totem-plugins-engine.c	2025-09-09 15:26:14.748670355 +0100
@@ -35,7 +35,7 @@
 
 #include <glib/gi18n.h>
 #include <glib.h>
-#include <girepository.h>
+#include <girepository/girepository.h>
 #include <libpeas/peas-activatable.h>
 #include <libpeas/peas-extension-set.h>
 
@@ -101,9 +101,9 @@ totem_plugins_engine_get_default (TotemO
 
 	g_return_val_if_fail (totem != NULL, NULL);
 
-	g_irepository_require (g_irepository_get_default (), "Peas", "1.0", 0, NULL);
-	g_irepository_require (g_irepository_get_default (), "PeasGtk", "1.0", 0, NULL);
-	g_irepository_require (g_irepository_get_default (), "Totem", TOTEM_API_VERSION, 0, NULL);
+	gi_repository_require (gi_repository_dup_default (), "Peas", "1.0", 0, NULL);
+	gi_repository_require (gi_repository_dup_default (), "PeasGtk", "1.0", 0, NULL);
+	gi_repository_require (gi_repository_dup_default (), "Totem", TOTEM_API_VERSION, 0, NULL);
 
 	paths = totem_get_plugin_paths ();
 
